tests/unit/amo/reducers/test_users.js

```diff
@@ -63,6 +63,22 @@ describe(__filename, () => {
 
       expect(currentUserID).toEqual(null);
     });
+
+    it('indexes a loaded user ID by username', () => {
```

Comment:
The only change you made to the reducer is to ensure that the usernames used for the key of `byUsername` is always lower case, so why do these two tests need to be added. Is this just missing coverage that you discovered while working on the patch?

Reply:
That was not covered before. If I revert my changes in the reducer, those tests will fail.

Reply:
Right, but they will only fail because you're mixing the case. If you hadn't done that then similar tests would have passed without your patch. You have a different test that verifies that it's case insensitive, and these tests don't describe themselves as testing that.

Reply:
Indeed, I don't know what I was thinking this morning but the test descriptions are not really accurate... Thanks for pointing this out.

I'll update them but I want to keep them because they test the logic to add users into the state. The other test is for the selector. This selector could have been updated to find a user by a username in a case insensitive manner (by iterating over the users for instance) but I chose not to do that because I really want to store usernames in lowercase too. Hence the two different kind of tests.

Reply:
I just thought of one last thing: Is it possible for there to be two users, with different userids and the same username but with different casing? E.g., a `Bob` and a `bob`? If so, wouldn't that present a potential issue with this implementation?

Reply:
Yes, I thought about that too but I assumed that would not be possible because the user API returned a user for a username with a different case (that's how I stumbled upon this bug actually).

Reply:
It has been confirmed on IRC: `username` is unique.

Reply:
... unique in a case insensitive way, see: https://github.com/mozilla/addons-server/blob/e4d1817c934cbd700c3700d67ffd08417f54d099/src/olympia/users/forms.py#L176-L179